src/main/java/org/mule/extension/http/internal/listener/HttpListener.java

```diff
@@ -111,6 +111,9 @@
   private static final HttpListenerErrorResponseBuilder NULL_ERROR_RESPONSE = new HttpListenerErrorResponseBuilder();
 
   @Inject
+  private TransformationService transformationService;
+
+  @Inject
   private MuleContext muleContext;
```

Comment:
org.mule.extension.http.internal.request.HttpRequestFactory#createRequestEntityFromPayload still uses is for transformation. If you replace that one with the TransformationService, pretty sure you can replace it here as well